src/controller/CHIPDeviceController.cpp

```diff
@@ -710,6 +707,23 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, co
                                        resolutionData);
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+static void StopSignalHandler(int signum)
+{
+    WiFiPAF::WiFiPAFLayer::GetWiFiPAFLayer()->Shutdown([](uint32_t id, WiFiPAF::WiFiPafRole role) {
+        switch (role)
+        {
+        case WiFiPAF::WiFiPafRole::kWiFiPafRole_Publisher:
+            DeviceLayer::ConnectivityMgr().WiFiPAFCancelPublish(id);
+            break;
+        case WiFiPAF::WiFiPafRole::kWiFiPafRole_Subscriber:
+            DeviceLayer::ConnectivityMgr().WiFiPAFCancelSubscribe(id);
+            break;
+        }
+    });
+}
```

Comment:
Signal handling not allowed directly in portable code. The signal handling must be in platform specific code and code here must remain fully portable.

Reply:
I will move them to src/platform/Linux/ConnectivityManagerImpl.cpp